### PR TITLE
Get /home Route 

### PR DESCRIPTION
**Pull request by RandSohail**
- Bring locations data
- Bring top 10 reviews 
- Bring main services  
- Test:
  - status code
  - JSON data
  - object contains 3 property => location, services, topTenReviews
  - Location Property contains with a length of 12
  - topTenReviews contain data less or equal to 10 all the data in it with a rate of 5.
  - services has all the main services with a length 10 and the first one with 'الحدادة' name.

### CODE REVIEW

**Review by abdulrahman-2020 — APPROVED** [review]

File: server/controllers/home/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    const locationFromDB = await Location.findAll();
+    const mainServices = await MainServices.findAll();
+    const reviewFromDB = await Review.findAll({
+      limit: 10,
+      where: {
+        rate: 5,
+      },
+    });
+    // eslint-disable-next-line max-len
+    response.json({ location: locationFromDB, topTenReviews: reviewFromDB, services: mainServices });
```
Comment (devstm) [inline server/controllers/home/index.js:16]: response must be json opject with msg and data!
const data = { location: locationFromDB, topTenReviews: reviewFromDB, services: mainServices }
response.json({ msg: "home page data" , data});

**Review by devstm — APPROVED** [review]

File: server/controllers/home/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  try {
+    const locationFromDB = await Location.findAll();
+    const mainServices = await MainServices.findAll();
+    const reviewFromDB = await Review.findAll({
+      limit: 10,
+      where: {
+        rate: 5,
+      },
+    });
+    // eslint-disable-next-line max-len
```
Comment (LinaYahya) [inline server/controllers/home/index.js:14]: Don't disable just wrap your line

File: server/controllers/home/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const { Location, Review, MainServices } = require('../../database/models');
+const { customError } = require('../errors');
+
+const Home = async (request, response, next) => {
+  try {
+    const locationFromDB = await Location.findAll();
+    const mainServices = await MainServices.findAll();
+    const reviewFromDB = await Review.findAll({
```
Comment (LinaYahya) [inline server/controllers/home/index.js:8]: here we're waiting for locations to get mainServivces after that we get top ten reviews, 
this id take a time while we're don't need to wait for location to get services as it don't depend on
So maybe we can use 
`Promise.all([])` OR `Promise.all Settled([])`

Read more about the difference [Promise all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)

File: server/routes/index.js [hunk 4]
```diff
@@ -1,7 +1,10 @@
 const router = require('express').Router();
+const { Home } = require('../controllers');
 
 router.get('/', async (req, res) => {
   res.json({ msg: 'hello' });
 });
 
+router.get('/home', Home);
```
Comment (LinaYahya) [inline server/routes/index.js:8]:
```suggestion
});

router.get('/home', getHomeData);
```

File: server/test/homeRoute.test.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
+const request = require('supertest');
+const app = require('../app');
+
```
Comment (LinaYahya) [inline server/test/homeRoute.test.js:3]: I suppose that you run build for test_url,
But this is locally
So we need here to run the build function 
```js
beforeAll(() => build)
```
and after all we need to close connection

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: server/controllers/home/index.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+const { Location, Review, MainServices } = require('../../database/models');
+const { customError } = require('../errors');
+
+const getHomeData = async (request, response, next) => {
+  try {
+    const data = await Promise.all([{
+      location: await Location.findAll(),
+      services: await MainServices.findAll(),
+      topTenReviews: await Review.findAll({
```
Comment (LinaYahya) [inline server/controllers/home/index.js:9]: promise.all built to receive an array of promises, What you're doing here you give it one element as an object that will wait for promises to finish

File: server/routes/index.js [hunk 7]
```diff
@@ -1,7 +1,10 @@
 const router = require('express').Router();
+const { getHomeData } = require('../controllers');
 
 router.get('/', async (req, res) => {
   res.json({ msg: 'hello' });
 });
 
+router.get('/home', getHomeData);
```
Comment (LinaYahya) [inline server/routes/index.js:8]: I would prefer if our endpoint was `/` instead of `/home`

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by AbdallahGot — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**